### PR TITLE
fix: tuning guide

### PR DESCRIPTION
**Pull request by zmstone**
1. add swap turn off
1. allow 2 million FDs for 1 million clients

### CODE REVIEW

File: en_US/tutorial/tune.md [hunk 1]
```diff
@@ -2,90 +2,129 @@
...
+This guide includes in general tuning suggestions for one EMQ X broker to serve about 1 million clients.
+
+## Turn off swap
+
+Linux swap partitions may cause nondeterministic memory latency to Erlang virutal machine,
+which in turn significantly affects the system stability.
+It is recommended to turn off swap permanently.
+
+To turn off swap immediately, execute command `sudo swapoff -a`.
+To turn off swap permanently, coment out the `swap` line in `/etc/fstab` and reboot the host
```
Comment (thalesmg) [inline en_US/tutorial/tune.md]:
```suggestion
To turn off swap permanently, comment out the `swap` line in `/etc/fstab` and reboot the host
```

File: en_US/tutorial/tune.md [hunk 2]
```diff
@@ -2,90 +2,129 @@
...
 ```
-Set the maximum number of file handles for the service in /etc/systemd/system.conf:
+
+Set the maximum number of file handles for the service in `/etc/systemd/system.conf`:
+
+```
+DefaultLimitNOFILE=2097152
+```
+
+### /usr/lib/systemd/system/emqx.service
```
Comment (thalesmg) [inline en_US/tutorial/tune.md]: Maybe

```suggestion
### /lib/systemd/system/emqx.service
```
?

I patch this file (without `/usr`) when tuning for the performance tests.

File: en_US/tutorial/tune.md [hunk 3]
```diff
@@ -2,90 +2,129 @@
...
+
 ```
 sysctl -w net.ipv4.ip_local_port_range='1000 65535'
 ```
- TCP Socket read/write buffer: 
+
+TCP Socket read/write buffer:
+
 ```
 sysctl -w net.core.rmem_default=262144
```
Comment (thalesmg) [inline en_US/tutorial/tune.md:87]: Small detail: a few of these are a bit different in [our CDK tuning](https://github.com/emqx/cdk-emqx-cluster/blob/6bc2af27e5510fba5e6ad5d63a5f141a68e99cf5/user_data/emqx_init.sh#L3-L17).

**Review by thalesmg — DISMISSED** [review]

File: en_US/tutorial/tune.md (hunk 2 above)
Comment (zmstone) [reply en_US/tutorial/tune.md]: it's /usr/lib in centos
i have listed both in the latest commit.

File: en_US/tutorial/tune.md (hunk 3 above)
Comment (zmstone) [reply en_US/tutorial/tune.md:87]: Thanks. theses are different.
```
net.core.rmem_default=262144000
net.core.wmem_default=262144000
net.core.rmem_max=262144000
net.core.wmem_max=262144000
net.ipv4.tcp_mem=378150000  504200000  756300000
```
but I'm not sure if they are generic enough for all

File: en_US/tutorial/tune.md [hunk 4]
```diff
@@ -2,90 +2,133 @@
 
 For EMQ X Message Server 4.x version, MQTT connection stress test reached 1.3 million on an 8-core, 32G memory CentOS server.
 
-Linux kernel parameters, network protocol stack parameters, Erlang virtual machine parameters, and EMQ X message server parameter settings required for the 1 million connection test are as follows:
+This guide includes in general tuning suggestions for one EMQ X broker to serve about 1 million clients.
+
+## Turn off swap
+
+Linux swap partitions may cause nondeterministic memory latency to Erlang virutal machine,
```
Comment (thalesmg) [inline en_US/tutorial/tune.md]:
```suggestion
Linux swap partitions may cause nondeterministic memory latency to Erlang virtual machine,
```

**Review by thalesmg — DISMISSED** [review]

**Pull request merged**